org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/editor/model/ResourceForIEditorInputFactory.java

```diff
@@ -107,6 +112,8 @@ protected Resource createResource(IStorage storage) throws CoreException {
 	}
 
 	protected Resource createResourceFor(IStorage storage) throws CoreException {
+		if (storage.getFullPath() == null) 
+			return null;
 		ResourceSet resourceSet = getResourceSet(storage);
 		URI uri = URI.createPlatformResourceURI(storage.getFullPath().toString(), true);
```

Comment:
Please remove the null guard above and to a second attempt in line 118 `URI.createPlatformResourceURI(storage.getFullPath().toString(), true);` with the name of the storage if the path is null. If both are null, I'd say we should throw an exception similar to the one in `createResource(IEditorInput)`.
Out of paranoia we could even try with storage.toString() and attempt to figure if this looks like a file name.

Reply:
I check for fullPath and name, but I am not sure if using storage.getName is really a good idea. In this case I loose the project information which for instance is required by xbase dsls to compute the Java classpath.